homeassistant/components/tuya/climate.py

```diff
@@ -0,0 +1,463 @@
+        _LOGGER.debug("climate add-> %s", dev_ids)
+        if not dev_ids:
+            return
+        entities = await hass.async_add_executor_job(
+            _setup_entities, hass, entry, dev_ids
+        )
+        async_add_entities(entities)
+
+    entry.async_on_unload(
+        async_dispatcher_connect(
+            hass, TUYA_DISCOVERY_NEW.format(DEVICE_DOMAIN), async_discover_device
+        )
+    )
+
+    device_manager = hass.data[DOMAIN][entry.entry_id][TUYA_DEVICE_MANAGER]
+    device_ids = []
+    for (device_id, device) in device_manager.device_map.items():
+        if device.category in TUYA_SUPPORT_TYPE:
+            device_ids.append(device_id)
+    await async_discover_device(device_ids)
+
+
+def _setup_entities(
+    hass: HomeAssistant, entry: ConfigEntry, device_ids: list[str]
+) -> list[Entity]:
+    """Set up Tuya Climate."""
+    device_manager = hass.data[DOMAIN][entry.entry_id][TUYA_DEVICE_MANAGER]
+    entities: list[Entity] = []
+    for device_id in device_ids:
+        device = device_manager.device_map[device_id]
+        if device is None:
+            continue
+        entities.append(TuyaHaClimate(device, device_manager))
+        hass.data[DOMAIN][entry.entry_id][TUYA_HA_DEVICES].add(device_id)
+    return entities
+
+
+class TuyaHaClimate(TuyaHaEntity, ClimateEntity):
+    """Tuya Switch Device."""
+
+    def __init__(
+        self, device: TuyaDevice, device_manager: TuyaDeviceManager
+    ) -> None:
+        """Init Tuya Ha Climate."""
+        super().__init__(device, device_manager)
+        if DPCODE_C_F in self.tuya_device.status:
+            self.dp_temp_unit = DPCODE_C_F
+        else:
+            self.dp_temp_unit = DPCODE_TEMP_UNIT_CONVERT
+
+    def get_temp_set_scale(self) -> int:
+        """Get temperature set scale."""
+        __dp_temp_set = DPCODE_TEMP_SET if self.__is_celsius() else DPCODE_TEMP_SET_F
+        __temp_set_value_range = json.loads(
+            self.tuya_device.status_range.get(__dp_temp_set).values
+        )
+        return __temp_set_value_range.get("scale")
+
+    def get_temp_current_scale(self) -> int:
+        """Get temperature current scale."""
+        __dp_temp_current = (
+            DPCODE_TEMP_CURRENT if self.__is_celsius() else DPCODE_TEMP_CURRENT_F
+        )
+        __temp_current_value_range = json.loads(
+            self.tuya_device.status_range.get(__dp_temp_current).values
+        )
+        return __temp_current_value_range.get("scale")
+
+    # Functions
+
+    def set_hvac_mode(self, hvac_mode):
+        """Set new target hvac mode."""
+        commands = []
+        if hvac_mode == HVAC_MODE_OFF:
+            commands.append({"code": DPCODE_SWITCH, "value": False})
+        else:
+            commands.append({"code": DPCODE_SWITCH, "value": True})
+
+        for tuya_mode, ha_mode in TUYA_HVAC_TO_HA.items():
+            if ha_mode == hvac_mode:
+                commands.append({"code": DPCODE_MODE, "value": tuya_mode})
```

Comment:
Should we allow more than one command or should we break the loop after the first match?

Reply:
fixed